include/seastar/websocket/common.hh

```diff
@@ -92,10 +92,16 @@ protected:
     public:
         connection_sink_impl(queue<buff_t>* data) : data(data) {}
 
+#if SEASTAR_API_LEVEL < 9
         virtual future<> put(net::packet d) override {
             net::fragment f = d.frag(0);
             return data->push_eventually(temporary_buffer<char>{std::move(f.base), f.size});
         }
+#else
+        virtual future<> put(temporary_buffer<char> buf) override {
+            return data->push_eventually(std::move(buf));
+        }
```

Comment:
Something like push_eventually will need to stabilize a span if it gets one.

Reply:
Perhaps a better approach is to push immediately, but resolve the future only when control flow allows. So at least we know a fast implementation is possible.